discovery-provider/src/queries/reactions.py

```diff
@@ -0,0 +1,37 @@
+from typing import List, Tuple, TypedDict
+
+from sqlalchemy.orm.session import Session
+from src.models.models import User
+from src.models.reaction import Reaction
+
+
+class ReactionResponse(TypedDict):
+    reaction_value: str
+    reaction_type: str
+    reacted_to: str
+    sender_user_id: int
+
+
+def get_reactions(
+    session: Session, transaction_ids: List[str], type: str
+) -> List[ReactionResponse]:
+    results: List[Tuple[Reaction, int]] = (
+        session.query(Reaction, User.user_id)
+        .join(User, User.wallet == Reaction.sender_wallet)
+        .filter(
+            Reaction.reacted_to.in_(transaction_ids),
+            Reaction.reaction_type == type,
+            User.is_current == True,
+        )
+        .all()
+    )
+
+    return [
```

Comment:
and then yeah we do list comp in this case but for loop in previous? just wondering if there is a pattern i should follow!

Reply:
that is a very good point :joy: I think the "pattern" being used here is "match neighbors" - since the other notifications were using normal `for` and `append` I think we both just blindly followed suit.

But you inspired me to look up the implications [and it's pretty interesting!](https://towardsdatascience.com/list-comprehensions-vs-for-loops-it-is-not-what-you-think-34071d4d8207)